### ISSUE TITLE
Version 2.0.0 unable to create database connection

### ISSUE THREAD

**Issue by kurawes** [incidentbot/incidentbot#589]
**Describe the bug**
I have incident-bot set up in AWS EKS and using Postgres RDS. Everything worked in version 1.12.2 (with some minor bug - status change was not recorded in postmortems), but with the newest 2.0.0 version getting the following error and kubernetes pod is crashlooping:

```
2024-09-20 12:16:56 [info     ] Starting task scheduler...    
2024-09-20 12:16:57 [info     ] Testing the database connection...
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 941, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UndefinedTable: relation "user" does not exist
LINE 2: FROM "user" 
             ^

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/app/main.py", line 183, in <module>
    startup_tasks()
  File "/app/main.py", line 68, in startup_tasks
    create_default_admin_user()
  File "/app/incidentbot/models/database.py", line 353, in create_default_admin_user
    user = session.exec(
           ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/sqlmodel/orm/session.py", line 66, in exec
    results = super().execute(
              ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 2362, in execute
    return self._execute_internal(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 2247, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/orm/context.py", line 305, in orm_execute_statement
    result = conn.execute(
             ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1418, in execute
    return meth(
           ^^^^^
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/sql/elements.py", line 515, in _execute_on_connection
    return connection._execute_clauseelement(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1640, in _execute_clauseelement
    ret = self._execute_context(
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1846, in _execute_context
    return self._exec_single_context(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1986, in _exec_single_context
    self._handle_dbapi_exception(
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 2355, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
  File "/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 941, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "user" does not exist
```

**Version**
2.0.0 via helm chart

**To Reproduce**

**Expected behavior**
Database connection working.

**Environment (please complete the following information):**
 - Platform: AWS EKS via helm chart

**Additional context**
Was there some new helm values option introduced that i might have missed? I did even drop the postgres database before installing the newest version. Even did a test where i recreated the database structure with 1.12.2 and installed 2.0.0 after that, but still getting the same error.

My helm chart setup (simplified):

```
configMap:
  create: true
  data:
    platform: slack
    digest_channel: incidents

    roles:
      incident_commander:
        description: "The Incident Commander is the decision maker during an incident."
        is_lead: true

    severities:
      P1_blocker: 'Blocker incident.'
      P2_critical: 'Critical.'
      P3_major: 'Important.'
      P4_minor: 'Minor issue.'
      P5_trivial: 'Trivial issue.'

    options:
      timezone: UTC
      meeting_link: 'https://meet.google.com/'
      skip_logs_for_user_agent:
        - kube-probe
        - ELB-HealthChecker/2.0

    integrations: 
      atlassian:
        confluence:
          enabled: true
          auto_create_postmortem: true
          space: XX
          parent: Postmortems
          template_id: 123456
        jira:
          enabled: true
          auto_create_issue: false
          project: XXX
          issue_types: ['Support', 'Bug', 'Service Request']
          priorities: ['Blocker', 'Critical', 'Major', 'Minor', 'Trivial']
          labels:
            - incident-management
            - incident-bot
          status_mapping:
            - incident_status: Investigating
              jira_status: Open
            - incident_status: Identified
              jira_status: Open
            - incident_status: Monitoring
              jira_status: Open
            - incident_status: Resolved
              jira_status: Completed

    links:
      - title: Incident Guide
        url: https://myatlassian.atlassian.net/Incidents
      - title: Postmortems
        url: https://myatlassian.atlassian.net/Postmortems

envFromSecret:
  enabled: true
  secretName: ib-secrets
ingress:
  enabled: true
  annotations:
    kubernetes.io/ingress.class: alb
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP": 80}, {"HTTPS":443}]'
    alb.ingress.kubernetes.io/scheme: internal
    alb.ingress.kubernetes.io/group.name: internal-ingress-group
    alb.ingress.kubernetes.io/ssl-redirect: "443"
    alb.ingress.kubernetes.io/target-type: ip
    cert-manager.io/cluster-issuer: null
  hosts:
    - host: incident-bot.mywebpage.com
      paths:
        - path: /
          pathType: Prefix
  tls: null
```

**Comment by echoboomer** [comment]
@kurawes thanks for reporting - this is related to the migrations that take place prior to starting the bot up. It would probably be best if an init job was added to handle this in the chart, so I'll look into adding that right now.

That being said, this is likely also a weakness in the docs which I will fix. In the meantime, if you're able to run the migrations against your database you should be able to get past this.

**Comment by kurawes** [comment]
Could you give me a bit of direction (a link) about the migrations? It's kinda hard to find the proper info. :)

**Comment by Kostavro** [comment]
Hey, I ran into the same issue today while updating to v2. I am managing the app through Helm. To resolve this I just changed the command of the Deployment/Pod to something like `sleep 1d`, and then ran `alembic upgrade head` - got that from `Makefile`. The schema was created then.

**Comment by echoboomer** [comment]
Thanks for the quick response @Kostavro - that's exactly what I was getting at. I'm going to push up a change shortly that will handle the migrations via the chart. Apologies for the omission.

**Comment by echoboomer** [comment]
Ok, so there's a lot going on here.

- The last error indicates the bot isn't even in the incident channel when it tries to send that message, which is odd because the bot should also be the first member. Can you confirm this is the case?
- I will look into the optional field issue.
- Can you share the content of your `values.yaml` file for Helm with anything sensitive stripped?

**Comment by echoboomer** [comment]
There was an issue with how the app sources the `config.yaml` file built from the Helm install. I'm going to push up a fix for this.

**Comment by echoboomer** [comment]
@kurawes just released `v2.0.4`, if you have time please try bumping to that chart version and seeing if your issues are resolved. Thanks.
